src/components/Rubric.js

```diff
@@ -1,177 +1,171 @@
+                {this.renderCriteriaForLevel(index + 1, compIndex)}
+              </tbody>
+            </table>
           </div>
         )
       })
     }
-  }
 
-  renderCriteriaForLevel(level, compIndex) {
-    if (this.props.assessment.rubricJSON.Competencies) {
-      return this.props.assessment.rubricJSON.Competencies[compIndex].Scales.map((scale, index) => {
-        // if the criteria level matches the level parameter, add the
-        // criteria component
-        return scale.Criteria.filter(criteria => criteria.level === level).map((criteria, index) => {
-          return <Criteria text={criteria.text} />
+    renderCompetencies() {
+      if (this.props.assessment.assessmentObject) {
+        return this.props.assessment.assessmentObject.rubricJSON.Competencies.map((comp, index) => {
+          var active = "";
+          if (index === this.state.activeCompetencyIndex) {
+            active = "active";
+          }
+          return (
+            <div className={`tab-pane ${active}`} key={uuidv1()} id="dashboard-2">
+              {this.getIsFetching() ? "" : this.renderLevels(index)}
+            </div>
+          )
+        }).sort((a , b) => {
+          return a.id - b.id;
         })
-      })
+      }
     }
-  }
 
-  renderLevels(compIndex) {
-    const levelNames = ['Initial', 'Approaching', 'Overtaking', 'Innovating'];
-    return levelNames.map((levelName, index) => {
-      return (
-        <div key={uuidv1()} className='col-md-3'>
-          <h3> {levelName} </h3>
-          <hr />
-          <table key={uuidv1()} className="table">
-            <tbody key={uuidv1()} >
-              {this.renderCriteriaForLevel(index + 1, compIndex)}
-            </tbody>
-          </table>
-        </div>
-      )
-    })
-  }
+    setActiveComp(index) {
+      this.setState({ activeCompetencyIndex: index })
+    }
 
-  getIsFetching() {
-    if (this.props.assessment == null || this.props.assessment.isFetching == null) {
-      return true;
+    renderCompetencyButtons() {
+      if (this.props.assessment.assessmentObject) {
+        return this.props.assessment.assessmentObject.rubricJSON.Competencies.map((comp, index) => {
+          var active = "";
+          if (index === this.state.activeCompetencyIndex) {
+            active = "active";
+          }
+          return (
+            <CompetencyButton
+              name={comp.name}
+              key={uuidv1()}
+              index={index}
+              isActive={active}
+              icon={this.icons[index]}
+              setActiveComp={this.setActiveComp.bind(this)}
+              />
+          )
+        }).sort((a , b) => {
+          return a.id - b.id;
+        })
+      }
     }
-    return this.props.assessment.isFetching;
-  }
 
+    getIsFetching() {
+      if (this.props.assessment == null || this.props.assessment.isFetching == null) {
+        return true;
+      }
+      return this.props.assessment.isFetching;
+    }
```

Comment:
could this whole fn be simplified to:

`return this.props.assessment.isFetching === null || this.props.assessment === null;` 

?

Reply:
well actually is the value you're returning a boolean or a null value?